### PR TITLE
feat: SCRUM-152 added modal window and updated route details page

### PR DESCRIPTION
**Pull request by AlexeyBezrebriy**
## Describe your changes

- I added FE for modal window.
image

- I fixed sorting of orders for route details page.

## Issue ticket code (and/or) and link

- [SCRUM-152](https://codecraftersintership.atlassian.net/browse/SCRUM-152)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/RouteDetails/hooks/useProcessAddresses.ts [hunk 1]
```diff
@@ -14,16 +14,29 @@ export const useProcessAddresses = (
...
 
   const processAddresses = useCallback(async () => {
     if (!route) return;
 
     setIsAddressesLoading(true);
     try {
-      const driverLocation = await geocodeAddress('New York');
+      const driverLocationData = await geocodeAddress('New York');
+
+      if (!driverLocationData) throw new Error('Driver location not found');
```
Comment (asdelay) [inline src/pages/RouteDetails/hooks/useProcessAddresses.ts]: I guess you should use i18n here

**Review by asdelay — APPROVED** [review]

File: src/pages/RouteDetails/hooks/useProcessAddresses.ts [hunk 2]
```diff
@@ -14,16 +14,32 @@ export const useProcessAddresses = (
...
+  driverAddresses: Address | undefined;
 } => {
   const [addresses, setAddresses] = useState<Address[]>([]);
+  const [driverAddresses, setDriverAddresses] = useState<Address>();
   const [isAddressesLoading, setIsAddressesLoading] = useState(false);
 
   const processAddresses = useCallback(async () => {
     if (!route) return;
 
+    const driverLocation = 'New York';
```
Comment (Parle648) [inline src/pages/RouteDetails/hooks/useProcessAddresses.ts]: move to constants

File: src/pages/RouteDetails/hooks/useProcessAddresses.ts [hunk 3]
```diff
@@ -14,16 +14,32 @@ export const useProcessAddresses = (
   processAddresses: () => Promise<void>;
   addresses: Address[];
   isAddressesLoading: boolean;
+  driverAddresses: Address | undefined;
 } => {
   const [addresses, setAddresses] = useState<Address[]>([]);
+  const [driverAddresses, setDriverAddresses] = useState<Address>();
   const [isAddressesLoading, setIsAddressesLoading] = useState(false);
```
Comment (Parle648) [inline src/pages/RouteDetails/hooks/useProcessAddresses.ts]: add type for useState

File: src/pages/RouteDetails/index.tsx [hunk 4]
```diff
@@ -23,15 +24,23 @@ import Button from '@/components/Button';
 import Header from '@/components/Header';
 import Loader from '@/components/Loader';
 import { useAppSelector } from '@/redux/hooks';
-import { RouteInform } from '@/types/route';
+import { Address, RouteInform } from '@/types/route';
 
 const RouteDetails: React.FC = () => {
+  const [open, setOpen] = useState(false);
```
Comment (Parle648) [inline src/pages/RouteDetails/index.tsx]: same

File: src/pages/RouteDetails/index.tsx [hunk 5]
```diff
@@ -23,15 +24,23 @@ import Button from '@/components/Button';
...
   const user = useAppSelector((state) => state.auth.user);
   const route = useAppSelector((state) => state.route.route);
 
+  const handleOpen = (): void => {
```
Comment (Parle648) [inline src/pages/RouteDetails/index.tsx]: I'd better make useToggle hook for this

**Review by Parle648 — APPROVED** [review]

**Pull request merged**